spec/models/prescription_drug_spec.rb

```diff
@@ -14,6 +14,52 @@
     it_behaves_like "a record that is deletable"
   end
 
+  describe ".prescribed_as_on" do
+    def remove_drug(prescription_drug, time)
+      prescription_drug.update(is_deleted: true, device_updated_at: time)
+    end
+
+    let!(:facility) { create(:facility) }
+    let!(:patient) { create(:patient) }
+    let!(:user) { create(:user) }
+    let!(:initial_visit_time) { Time.parse "01 Jan 2020 14:30" }
+    let!(:latest_visit_time) { Time.parse "15 Jan 2020 14:30" }
+    let!(:prescription_drugs) do
+      Timecop.freeze(initial_visit_time) do
+        create_list(:prescription_drug, 2, is_protocol_drug: true, patient: patient, facility: facility, user: user)
```

Comment:
Why do we have to create and assign a facility and user? Ideally, the factory should create em if missing. If we _have_ to assign em, could we just use `patient.registration_*` to save on some db operations?